File: src/time.rs
use std::process::{Stdio};
use std::time::{Instant};
use crate::ProcessCommand;

// Function to measure and print the execution time of a command
pub fn time_command(arguments: Vec<String>) {
    if arguments.is_empty() {
        eprintln!("Usage: time [COMMAND] [ARGS]...");
        return;
    }

    let command_name = &arguments[0];
    let args = &arguments[1..];

    let start_time = Instant::now();

    let status = ProcessCommand::new(command_name)
        .args(args)
        .stdout(Stdio::inherit())
        .stderr(Stdio::inherit())
        .status();

    let duration = start_time.elapsed();

    match status {
        Ok(status) => {
            println!("\nreal: {:.2?}", duration);
            // Note: `user` and `sys` times are typically platform-specific and not available directly in Rust's standard library.
            println!("user: n/a");
            println!("sys: n/a");

            if !status.success() {
                eprintln!("Command {} failed with status {}", command_name, status);
            }
        }
        Err(err) => {
            eprintln!("Failed to execute command {}: {}", command_name, err);
        }
    }
}

File: src/time.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"time"
)

// Function to measure and print the execution time of a command
func timeCommand(arguments []string) {
	if len(arguments) == 0 {
		fmt.Fprintln(os.Stderr, "Usage: time [COMMAND] [ARGS]...")
		return
	}

	commandName := arguments[0]
	args := arguments[1:]

	startTime := time.Now()

	cmd := exec.Command(commandName, args...)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr

	err := cmd.Run()

	duration := time.Since(startTime)

	if err == nil {
		fmt.Printf("\nreal: %.2fs\n", duration.Seconds())
		fmt.Println("user: n/a")
		fmt.Println("sys: n/a")
	} else {
		fmt.Printf("\nreal: %.2fs\n", duration.Seconds())
		fmt.Println("user: n/a")
		fmt.Println("sys: n/a")
		fmt.Fprintf(os.Stderr, "Failed to execute command %s: %v\n", commandName, err)
	}
}
